lib/fluent/plugin/out_cloudwatch_logs.rb

```diff
@@ -261,14 +261,32 @@ def create_log_stream(group_name, stream_name)
     def log_group_exists?(group_name)
       if @sequence_tokens[group_name]
         true
-      elsif @logs.describe_log_groups.log_groups.any? {|i| i.log_group_name == group_name }
+      elsif (log_group = find_log_group(group_name))
         @sequence_tokens[group_name] = {}
         true
       else
         false
       end
     end
 
+    def find_log_group(group_name)
```

Comment:
This can be done in this way too:

```ruby
log_group = nil
response = @logs.describe_log_groups(...)
response.each do |page|
  log_group = page.log_groups.find {|i| i.log_group_name == group_name}
end

log_group
```